datasafe-simple-adapter/datasafe-simple-adapter-impl/src/main/java/de/adorsys/datasafe/simple/adapter/impl/SwitchablePathEncryptionImpl.java

```diff
@@ -0,0 +1,54 @@
+package de.adorsys.datasafe.simple.adapter.impl;
+
+import de.adorsys.datasafe.directory.api.profile.keys.PrivateKeyService;
+import de.adorsys.datasafe.encrypiton.api.pathencryption.encryption.SymmetricPathEncryptionService;
+import de.adorsys.datasafe.encrypiton.api.types.UserIDAuth;
+import de.adorsys.datasafe.encrypiton.impl.pathencryption.PathEncryptionImpl;
+import de.adorsys.datasafe.types.api.resource.Uri;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+
+@Slf4j
+public class SwitchablePathEncryptionImpl extends PathEncryptionImpl {
+
+    public static final String NO_BUCKETPATH_ENCRYPTION = "SC-NO-BUCKETPATH-ENCRYPTION";
+
+    private static boolean withPathEncryption = checkIsPathEncryptionToUse();
+
+    @Inject
+    public SwitchablePathEncryptionImpl(SymmetricPathEncryptionService bucketPathEncryptionService, PrivateKeyService privateKeyService) {
+        super(bucketPathEncryptionService, privateKeyService);
+    }
+
+    public static boolean checkIsPathEncryptionToUse() {
+        if (System.getProperty(NO_BUCKETPATH_ENCRYPTION) != null) {
```

Comment:
maybe extracting local variable before if 
`String value = System.getProperty(NO_BUCKETPATH_ENCRYPTION);`
could be better - no double access to system-properties